src/hotspot/share/cds/filemap.cpp

```diff
@@ -2141,7 +2139,7 @@ bool FileMapInfo::map_heap_region_impl() {
                               addr, _mapped_heap_memregion.byte_size(), r->read_only(),
                               r->allow_exec());
   if (base == nullptr || base != addr) {
-    dealloc_heap_region();
+    handle_failed_mapping();
```

Comment:
`handle_failed_mapping()` is called for several reasons where we are not able to use the archive heap:
- mmap failed for the `hp` or `bm` regions
- CRC check failure

So I think the wording of "mapping" is too specific and doesn't cover all cases (and also precludes the possibility of other failures in the future.

Also "handle" could imply a possibility of recovering from the failure, which isn't the case.

I think a more neutral name would be better. Maybe `release_archive_space()` in contrast of `alloc_archive_space` (or even `allocate_archive_space`, as we prefer proper words than short hands).

Reply:
> I think a more neutral name would be better. Maybe `release_archive_space()` in contrast of `alloc_archive_space` (or even `allocate_archive_space`, as we prefer proper words than short hands).

"on_load_error"?

Reply:
To be honest I don't like these names either and am open for suggestions. I liked @tstuefe's suggestion for `on_cds_post_load` and `on_cds_load_error` but probably name them `on_archive_heap_post_load` and `on_archive_heap_load_error`.
Also in CDS code "load" is currently associated with the mechanism used by non-G1 collecters to read the archive space into the heap memory and "map" is associated with the G1 collector's mechanism to mmap the archive space. For instance FileMapInfo::map_or_load_heap_region.
Going forward I think we would need to get rid of this distinction, at least in public APIs and just use the term "load". Internally the "loading" may be achieved through IO read or mmap-ing the archive space.
@iklam what do you think?

Reply:
I will also update `alloc_archive_space` to `allocate_archive_space`. Thanks @iklam for pointing that out.